docs/source/user/file_formats.md

```diff
@@ -1,162 +1,151 @@
-
 # File and output formats
 
-## Overview
-
-When working with code and data, you will encounter data and files in a wide
-variety of formats. JupyterLab provides a unified architecture for working with
-data. This model applies whether the data is in a file, or is provided by a
-Kernel as output in a notebook or code console.
+JupyterLab provides a unified architecture for viewing and editing data in a
+wide variety of formats. This model applies whether the data is in a file or is
+provided by a kernel as rich cell output in a notebook or code console.
 
 For files, the data format is detected by the extension of the file. A single
```

Comment:
It may or may not be worth mentioning here, but it can also be determined by the name of the file (such as "Makefile" or "Dockerfile")

Reply:
I think it's worth mentioning.

Reply:
I didn't realize that. Nice!

Reply:
does it recognize the filename in cases where there is no extension?

Reply:
So far as I know.